### ISSUE TITLE
Build failure (GHC 8 related?)

### ISSUE THREAD

**Issue by deepfire** [cgaebel/stm-conduit#37]
The build fails on GHC 8:
```
[4 of 5] Compiling Data.Conduit.Async.Composition ( Data/Conduit/Async/Composition.hs, dist/build/Data/Conduit/Async/Composition.o )

Data/Conduit/Async/Composition.hs:207:42: error:
    Not in scope: type constructor or class ‘Constraint’
```
Full build log:
```
these derivations will be built:
  /nix/store/jyz1m745y8nhxk7l2h05c2p2arihq42p-stm-conduit-2.7.0.drv
building path(s) ‘/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0’
setupCompilerEnvironmentPhase
Build with /nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204.
unpacking sources
unpacking source archive /nix/store/acbdmm448dhnw55qnb4f3qbdi5n1haky-stm-conduit-2.7.0.tar.gz
source root is stm-conduit-2.7.0
setting SOURCE_DATE_EPOCH to timestamp 1449558767 of file stm-conduit-2.7.0/test/Test.hs
patching sources
Run jailbreak-cabal to lift version restrictions on build inputs.
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-stm-conduit-2.7.0.drv-0/package.conf.d -j1 -threa
ded
[1 of 1] Compiling Main             ( Setup.lhs, /tmp/nix-build-stm-conduit-2.7.0.drv-0/Main.o
)
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.
0 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --with-gcc=gcc --package-db=/tmp/nix-build-
stm-conduit-2.7.0.drv-0/package.conf.d --ghc-option=-optl=-Wl,-rpath=/nix/store/nwjx03jf33pb5lg
y9bcsqv717wxfbvsk-stm-conduit-2.7.0/lib/ghc-8.0.0.20160204/stm-conduit-2.7.0 --enable-split-obj
s --disable-library-profiling --disable-profiling --enable-shared --enable-library-vanilla --en
able-executable-dynamic --disable-tests
Configuring stm-conduit-2.7.0...
Dependency async -any: using async-2.1.0
Dependency base -any: using base-4.9.0.0
Dependency cereal -any: using cereal-0.5.1.0
Dependency cereal-conduit -any: using cereal-conduit-0.7.2.5
Dependency conduit -any: using conduit-1.2.6.2
Dependency conduit-combinators -any: using conduit-combinators-1.0.3.1
Dependency conduit-extra -any: using conduit-extra-1.1.10.1
Dependency directory -any: using directory-1.2.5.0
Dependency ghc-prim -any: using ghc-prim-0.5.0.0
Dependency lifted-async -any: using lifted-async-0.8.0.1
Dependency lifted-base -any: using lifted-base-0.2.3.6
Dependency monad-control -any: using monad-control-1.0.0.5
Dependency monad-loops -any: using monad-loops-0.4.3
Dependency resourcet -any: using resourcet-1.1.7.2
Dependency stm -any: using stm-2.4.4.1
Dependency stm-chans -any: using stm-chans-3.0.0.4
Dependency transformers -any: using transformers-0.5.1.0
Dependency void -any: using void-0.7.1
Using Cabal-1.23.1.0 compiled by ghc-8.0
Using compiler: ghc-8.0.0.20160204
Using install prefix:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0
Binaries installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/bin
Libraries installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/lib/ghc-8.0.0.20160204/stm-condui
t-2.7.0
Private binaries installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/libexec
Data files installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/share/x86_64-linux-ghc-8.0.0.2016
0204/stm-conduit-2.7.0
Documentation installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/share/doc/x86_64-linux-ghc-8.0.0.
20160204/stm-conduit-2.7.0
Configuration files installed in:
/nix/store/nwjx03jf33pb5lgy9bcsqv717wxfbvsk-stm-conduit-2.7.0/etc
No alex found
Using ar found on system at:
/nix/store/f9aj81zi2wam2dmnc30jhlr2lkwvgp49-binutils-2.23.1-dev/bin/ar
No c2hs found
No cpphs found
Using gcc version 4.9.3 given by user at:
/nix/store/2j5gi3bwsrpyfacivc28rlgiak2mdbxg-gcc-wrapper-4.9.3/bin/gcc
Using ghc version 8.0.0.20160204 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/ghc
Using ghc-pkg version 8.0.0.20160204 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.17.0 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/hpc
Using hsc2hs version 0.67 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/yrj2if1z9k739b92n4la852nvzlkz1h2-hscolour-1.24/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/2j5gi3bwsrpyfacivc28rlgiak2mdbxg-gcc-wrapper-4.9.3/bin/ld
No lhc found
No lhc-pkg found
No pkg-config found
Using strip version 2.23 found on system at:
/nix/store/f9aj81zi2wam2dmnc30jhlr2lkwvgp49-binutils-2.23.1-dev/bin/strip
Using tar found on system at:
/nix/store/hmfmqrjv73f201z9al1sgsbzkih39cvj-gnutar-1.28/bin/tar
No uhc found
building
Building stm-conduit-2.7.0...
Preprocessing library stm-conduit-2.7.0...
[1 of 5] Compiling Data.Conduit.TQueue ( Data/Conduit/TQueue.hs, dist/build/Data/Conduit/TQueue.o )
[2 of 5] Compiling Data.Conduit.TMChan ( Data/Conduit/TMChan.hs, dist/build/Data/Conduit/TMChan.o )
[3 of 5] Compiling Data.Conduit.Utils ( Data/Conduit/Utils.hs, dist/build/Data/Conduit/Utils.o )
[4 of 5] Compiling Data.Conduit.Async.Composition ( Data/Conduit/Async/Composition.hs, dist/build/Data/Conduit/Async/Composition.o )

Data/Conduit/Async/Composition.hs:207:42: error:
    Not in scope: type constructor or class ‘Constraint’
builder for ‘/nix/store/jyz1m745y8nhxk7l2h05c2p2arihq42p-stm-conduit-2.7.0.drv’ failed with exit code 1
error: build of ‘/nix/store/jyz1m745y8nhxk7l2h05c2p2arihq42p-stm-conduit-2.7.0.drv’ failed
```

**Comment by deepfire** [comment]
The fix is available in #38

**Comment by cgaebel** [comment]
Fixed
